chemprop/utils.py

```diff
@@ -522,12 +532,17 @@ def save_smiles_splits(data_path: str,
 
         split_indices = []
         for smiles in dataset.smiles():
-            split_indices.append(indices_by_smiles.get(tuple(smiles)))
+            split_indices.append(indices_by_smiles.get(tuple(smiles),-1))
             split_indices = sorted(split_indices)
+        if -1 in split_indices:
```

Comment:
A couple of suggestions here:
1. This section could probably be skipped entirely if the pickle file is not going to be generated
2. You could break the for loop when you detect the error condition similar to the earlier check
3. You didn't change this part, but I don't see why the list has to be sorted on every iteration of the loop - it could be moved outside of the for loop so the `split_indices` only needs to be sorted once

E.g.
```
if not pickle_warning_flag:  # "if save_smiles_splits" if you rename based on my other comment
    split_indices = []
    for smiles in dataset.smiles():
        indices = indices_by_smiles.get(tuple(smiles))
        if indices is None:
            pickle_warning_flag = True
            # print warning
            break
        split_indices.append(indices)
    else:  # runs only if "break" is not encountered
        split_indices.sort()  # sort in-place once
        all_split_indices.append(split_indices)
```